x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/ProcessResultsParser.java

```diff
@@ -3,31 +3,41 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.xpack.ml.job.process.autodetect.output;
+package org.elasticsearch.xpack.ml.process;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
 import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.common.xcontent.XContentFactory;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.common.xcontent.XContentType;
-import org.elasticsearch.xpack.ml.job.results.AutodetectResult;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Iterator;
+import java.util.Objects;
 
 
 /**
- * Parses the JSON output of the autodetect program.
+ * Parses the JSON output of a process.
  * <p>
- * Expects an array of buckets so the first element will always be the
+ * Expects an array of objects so the first element will always be the
  * start array symbol and the data must be terminated with the end array symbol.
  */
-public class AutodetectResultsParser {
-    public Iterator<AutodetectResult> parseResults(InputStream in) throws ElasticsearchParseException {
+public class ProcessResultsParser<T> {
+
+    private static final Logger LOGGER = LogManager.getLogger(ProcessResultsParser.class);
```

Comment:
I think the tradition is lowercase `logger` for static loggers, as the object is not immutable.